hibernate-core/src/main/java/org/hibernate/type/descriptor/java/LocalDateTimeJavaType.java

```diff
@@ -97,12 +97,12 @@ public <X> X unwrap(LocalDateTime value, Class<X> type, WrapperOptions options)
 
 		if ( java.sql.Date.class.isAssignableFrom( type ) ) {
 			Instant instant = value.atZone( ZoneId.systemDefault() ).toInstant();
-			return (X) java.sql.Date.from( instant );
+			return (X) new java.sql.Date( instant.toEpochMilli() );
 		}
 
 		if ( java.sql.Time.class.isAssignableFrom( type ) ) {
 			Instant instant = value.atZone( ZoneId.systemDefault() ).toInstant();
-			return (X) java.sql.Time.from( instant );
+			return (X) new java.sql.Time( instant.toEpochMilli() % 86_400_000 );
```

Comment:
Are you sure this is correct?

You're calling this constructor, right:

```java
    /**
     * Constructs a {@code Time} object using a milliseconds time value.
     *
     * @param time milliseconds since January 1, 1970, 00:00:00 GMT;
     *             a negative number is milliseconds before
     *               January 1, 1970, 00:00:00 GMT
     */
    public Time(long time) {
        super(time);
    }
```

Reply:
@gavinking If you are asking me, then answer is no. I am very sure (all the time) that use of remainder is wrong. It is simply disregarding the fact that time offset at some particular moment may be (and in often is) different than one on 1st January 1970. 
I've created test case `org.hibernate.orm.test.mapping.type.java.GoodBadUglyTest ` with three time zones for summer solstice 2024 at midnight  - 'good' one will pass, 'bad' will fail. Third one is called 'bad', because it can not pass because (unless you are using time zone data 2024b) 1st of January 1970  does not have first hour of day.

Reply:
@gavinking No, I am not sure at all that using remainder is correct. It is disregarding proper time offset.

Reply:
I mean that quoted javadoc does not suggest that you are required to strip off the date part. It appears to say the opposite.

Reply:
Wait, is this something @mbladel asked for? Marco are you sure about that?

Reply:
Hurm, OK, I guess it's not very clear. 🤷‍♂️

Reply:
That is main reason why I always believed that java.time was much more impprtant new feature of JDK 8 than streams, lambdas, ... :-)

Reply:
Just added suggestion how epoch time in milliseconds can be converted to `java.sql.Time` with (hidden) date part set to January 1st 1970, and time part to local time corresponding to original time in milliseconds and default time zone